Reporting/ad-extensions-reporter.js

```diff
@@ -0,0 +1,395 @@
+    )
+    for (var i = 0, n = sheets.length; i < n; i++) {
+        if (extensionTabNames.indexOf(sheets[i].getName()) !== -1) {
+            spreadsheet.deleteSheet(sheets[i]);
+        }
+    }
+
+    // Get report
+    Logger.log("Getting report");
+    if (typeof includeExtensionsWithZeroImpressions === 'undefined') {
+        Logger.log("Warning: options variable includeExtensionsWithZeroImpressions not found. " +
+            "Continuing, and by default including extensions that have had no impressions today."
+        );
+        impressionsCondition = '';
+    } else {
+        if (includeExtensionsWithZeroImpressions === true) {
+            var impressionsCondition = '';
+        } else if (includeExtensionsWithZeroImpressions === false) {
+            var impressionsCondition = 'WHERE Impressions > 0 ';
+        } else {
+            throw ("Value of options variable includeExtensionsWithZeroImpressions must be a boolean, either true or false. " +
+                "Currently, it is: " + includeExtensionsWithZeroImpressions)
+        }
+    }
+
+    var placeholderFeedItemReport = AdsApp.report(
+        'SELECT PlaceholderType, AttributeValues, DisapprovalShortNames ' +
+        'FROM PLACEHOLDER_FEED_ITEM_REPORT ' +
+        impressionsCondition +
+        'DURING TODAY'
+    );
+    var reportRows = placeholderFeedItemReport.rows();
+
+    // Process report data, to prepare for outputting to the spreadsheet
+    Logger.log("Processing report data");
+    var extensionPlaceholderTypeIds = Object.keys(extensionPlaceholderTypeIdToFieldIdsObjectMapping);
+
+    var extensionTypeFeedItemCounts = {};
+    for (var i = 0, n = extensionPlaceholderTypeIds.length; i < n; i++) {
+        extensionTypeFeedItemCounts[extensionPlaceholderTypeIds[i]] = 0;
+    }
+    var jsonAttributeValuesByExtensionType = {};
+    var disapprovalShortNamesByExtensionType = {};
+    var duplicateAttributeValuesCounts = {};
+
+    while (reportRows.hasNext()) {
+        var row = reportRows.next();
+
+        var placeholderTypeId = row['PlaceholderType']
+        if (extensionPlaceholderTypeIds.indexOf(placeholderTypeId) === -1) {
+            continue;
+        }
+        if (Object.keys(jsonAttributeValuesByExtensionType).indexOf(placeholderTypeId) === -1) {
+            jsonAttributeValuesByExtensionType[placeholderTypeId] = [];
+            disapprovalShortNamesByExtensionType[placeholderTypeId] = [];
+        }
+
+        var jsonAttributeValues = row['AttributeValues'];
+        var jsonDisapprovalShortNames = row['DisapprovalShortNames'];
+        if ('' === jsonDisapprovalShortNames.toString()) {
+            var disapprovalShortNames = [];
+        } else {
+            var disapprovalShortNames = JSON.parse(jsonDisapprovalShortNames);
+        };
+        if (jsonAttributeValuesByExtensionType[placeholderTypeId].indexOf(jsonAttributeValues) === -1) {
+            jsonAttributeValuesByExtensionType[placeholderTypeId].push(jsonAttributeValues);
```

Comment:
As this data is linked together (as in you have one jsonAttributeValues that maps to an array of disapprovalShortNames), rather than having two arrays, each containing one, and being intrinsically linked (aka relying on the fact that you always add to them at the same time, so they are always in the same order), you could have one thing e.g. `extensions`
```javascript
if (Object.keys(extensions[placeholderTypeId]).indexOf(jsonAttributeValues) === -1) {
            extensions[placeholderTypeId][jsonAttributeValues] = disapprovalShortNames ;
}
```
This would also simplify some of the code later on (not having to look up the index of the json, and then read that index from the disapproval names)

Reply:
Great idea, I like this much more now! I've also now included what was `duplicateAttributeValuesCounts` into this new object.